base/math.jl

```diff
@@ -851,26 +851,59 @@ minmax(x::T, y::T) where {T<:AbstractFloat} = min(x, y), max(x, y)
 
 _isless(x::Float16, y::Float16) = signbit(widen(x) - widen(y))
 
+const _minmaxtypes = [(:f64, :Float64, :double), (:f32, :Float32, :float)] #TODO: Add a way to detect native f16
+const _minmaxops = [(:min, :minimum), (:max, :maximum)]
+for op in _minmaxops
+    jlop,llvmop = op
+    for type in _minmaxtypes
+        fntyp, jltyp, llvmtyp = type
+        @eval @inline @assume_effects :total function $(Symbol("llvm_", jlop))(x::$jltyp,y::$jltyp)
+                Base.llvmcall(
+                    ($"""declare $llvmtyp @llvm.$llvmop.$fntyp($llvmtyp,$llvmtyp)
+                        define $llvmtyp @entry($llvmtyp,$llvmtyp) #0 {
+                        2:
+                            %3 = call $llvmtyp @llvm.$llvmop.$fntyp($llvmtyp %0, $llvmtyp %1)
```

Comment:
We export these normally through intrinsics. Compare `Core.Intrinsics.sqrt_llvm`

Reply:
The issue is that not all backends support it. Do we then call min from libm if someone hits that? Or just let llvm throw it's lowering error because they are doing wrong stuff anyway?

Reply:
Lowering to libm is fine by me.

Reply:
I tried to do more complicated lowering at some-point, and in theory we could lower back to Julia, but that is overcomplicating things.

Reply:
Actually lowering to libm is wrong, because libm min is different from ours because it doesn't respect the NaN handling we do.

Reply:
Then see `injectCRT` and `julia__gnu_h2f_ieee` to how we handle the compiler intrinsics. LLVM inserts those late during the backend.

Reply:
I would just implement min in C like we do with fma